### PR TITLE
[PTSBE] Integration tests with edge cases

### PR DESCRIPTION
**Pull request by sacpis**
Addressing feedback received from @taalexander for PR #3942. Adding more edge cases for the integration tests.

### CODE REVIEW

File: python/tests/ptsbe/conftest.py [hunk 1]
```diff
@@ -31,5 +22,31 @@ def density_matrix_target():
...
+
+
+@pytest.fixture
+def phase_flip_kernel_fixture():
+    return phase_flip_kernel
+
+
+@pytest.fixture
+def cnot_echo_kernel():
+    return cnot_echo
```
Comment (taalexander) [inline python/tests/ptsbe/conftest.py]: Could the kernels just be defined within and drop `.kernels`? If that doesn't work having them as fixtures might be redundant.

File: python/tests/ptsbe/test_errors.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+def test_ptsbe_broadcast_empty_args_returns_empty_list():
+    noise = cudaq.NoiseModel()
+    noise.add_all_qubit_channel("ry", cudaq.DepolarizationChannel(0.01))
+    results = cudaq.ptsbe.sample(
+        rotation_kernel,
+        [],
+        noise_model=noise,
+        shots_count=10,
+    )
+    assert results == []
```
Comment (taalexander) [inline python/tests/ptsbe/test_errors.py]: Can the kernels be consumed as fixtures? Similarily, noise models could be used in this way but not required.

**Review by taalexander — APPROVED** [review]
Thanks, should we split the existing tests in `test_ptsbe` out to the matching test file you created deleting tests if there are duplicates?

File: python/tests/ptsbe/conftest.py (hunk 1 above)
Comment (sacpis) [reply python/tests/ptsbe/conftest.py]: Fixed in a0fd4e2.

File: python/tests/ptsbe/test_errors.py (hunk 2 above)
Comment (sacpis) [reply python/tests/ptsbe/test_errors.py]: Fixed in a0fd4e2.

**Comment by sacpis** [comment]
> Thanks, should we split the existing tests in `test_ptsbe` out to the matching test file you created deleting tests if there are duplicates?

Fixed in [2c8821a](https://github.com/NVIDIA/cuda-quantum/pull/3944/commits/2c8821a3c4e30f699ee9f65628e7fad137fff912).

File: python/tests/ptsbe/test_ptsbe.py [hunk 3]
```diff
@@ -5,33 +5,9 @@
 # This source code and the accompanying materials are made available under     #
```
Comment (taalexander) [inline python/tests/ptsbe/test_ptsbe.py]: There's only one test here at this point. We should likely just remove this module.

File: python/tests/ptsbe/test_execution_data.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+def bell():
+    q = cudaq.qvector(2)
+    h(q[0])
+    x.ctrl(q[0], q[1])
+    mz(q)
+
+
+@pytest.fixture
+def bell_kernel():
+    return bell
```
Comment (taalexander) [inline python/tests/ptsbe/test_execution_data.py]: The same kernels are now duplicated in most tests? Why not just put them in conftest?

**Review by taalexander — CHANGES_REQUESTED** [review]
A few small changes left.

File: python/tests/ptsbe/test_ptsbe.py (hunk 3 above)
Comment (sacpis) [reply python/tests/ptsbe/test_ptsbe.py]: Fixed in f27ae82.

File: python/tests/ptsbe/test_execution_data.py (hunk 4 above)
Comment (sacpis) [reply python/tests/ptsbe/test_execution_data.py]: Fixed in f27ae82.
